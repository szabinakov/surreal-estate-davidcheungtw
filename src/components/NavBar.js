import React from "react";
import { Link } from "react-router-dom";
import FacebookLogin from "react-facebook-login";
import PropTypes from "prop-types";
import logo from "../images/logo.png";
import "../styles/navbar.css";

const NavBar = ({ menu, userID, onLogin, onLogout }) => {
  let classVP = "navbar-links-item";
  let classAP = "navbar-links-item";
  let classSP = "navbar-links-item";

  switch (menu) {
    case "classVP":
      classVP = `navbar-links-item ${menu}`;
      break;
    case "classAP":
      classAP = `navbar-links-item ${menu}`;
      break;
    default:
      classSP = `navbar-links-item ${menu}`;
  }

  return (
    <div className="navbar">
      <ul className="navbar-links">
        <img src={logo} className="navbar-logo" alt="logo" />
        <li className={classVP}>
          <Link className="item" to="/">
            View Properties
          </Link>
        </li>
        <li className={classAP}>
          <Link className="item" to="add-property">
            Add a Property
          </Link>
        </li>
        <li className={classSP}>
          <Link className="item" to="saved-properties">
            Saved Properties
          </Link>
        </li>
      </ul>

      <div className="facebook-login">
        {userID ? (
          <button type="submit" className="button-sign-out" onClick={onLogout}>
            Sign Out
          </button>
        ) : (
          <FacebookLogin
            appId="613942867273925"
            callback={onLogin}
            icon="fa-facebook"
            textButton="   Login"
          />
        )}
      </div>
    </div>
  );
};

export default NavBar;

NavBar.propTypes = {
  menu: PropTypes.string.isRequired,
  userID: PropTypes.string.isRequired,
  onLogin: PropTypes.func.isRequired,
  onLogout: PropTypes.func.isRequired,
};
