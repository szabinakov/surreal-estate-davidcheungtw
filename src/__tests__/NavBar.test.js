import React from "react";
import { MemoryRouter } from "react-router-dom";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import NavBar from "../components/NavBar";

describe("NavBar", () => {
  const fbScript = document.createElement("script");
  fbScript.id = "facebook-jssdk";
  document.body.appendChild(fbScript);

  it("Renders as expected", () => {
    const rendered = renderer.create(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>
    );

    expect(rendered).toMatchSnapshot();
  });

  it("Assert all menu bar selections", () => {
    render(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>
    );
    const img = screen.getByRole("img");

    expect(img).toBeInstanceOf(HTMLImageElement);
    expect(screen.getByText(/View Properties/i)).toBeInstanceOf(
      HTMLAnchorElement
    );
    expect(screen.getByText(/Add a Property/i)).toBeInstanceOf(
      HTMLAnchorElement
    );
    expect(screen.getByText(/Saved Properties/i)).toBeInstanceOf(
      HTMLAnchorElement
    );
    expect(screen.getByText(/Login/i)).toBeInstanceOf(HTMLButtonElement);
  });
});
