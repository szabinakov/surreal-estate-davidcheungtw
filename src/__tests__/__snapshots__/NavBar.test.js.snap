// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Renders as expected 1`] = `
<div
  className="navbar"
>
  <ul
    className="navbar-links"
  >
    <img
      alt="logo"
      className="navbar-logo"
      src="logo.png"
    />
    <li
      className="navbar-links-item"
    >
      <a
        className="item"
        href="/"
        onClick={[Function]}
      >
        View Properties
      </a>
    </li>
    <li
      className="navbar-links-item"
    >
      <a
        className="item"
        href="/add-property"
        onClick={[Function]}
      >
        Add a Property
      </a>
    </li>
    <li
      className="navbar-links-item undefined"
    >
      <a
        className="item"
        href="/saved-properties"
        onClick={[Function]}
      >
        Saved Properties
      </a>
    </li>
  </ul>
  <div
    className="facebook-login"
  >
    <span
      style={
        Object {
          "transition": "opacity 0.5s",
        }
      }
    >
      <link
        href="https://maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css"
        rel="stylesheet"
      />
      <button
        className="kep-login-facebook metro"
        onClick={[Function]}
        type="button"
      >
        <i
          className="fa fa-facebook"
        />
           Login
      </button>
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": Array [
              Array [
                9,
                ".kep-login-facebook{font-family:Helvetica,sans-serif;font-weight:700;-webkit-font-smoothing:antialiased;color:#fff;cursor:pointer;display:inline-block;font-size:calc(.27548vw + 12.71074px);text-decoration:none;text-transform:uppercase;transition:background-color .3s,border-color .3s;background-color:#4c69ba;border:calc(.06887vw + .67769px) solid #4c69ba;padding:calc(.34435vw + 13.38843px) calc(.34435vw + 18.38843px)}.kep-login-facebook.small{padding:calc(.34435vw + 3.38843px) calc(.34435vw + 8.38843px)}.kep-login-facebook.medium{padding:calc(.34435vw + 8.38843px) calc(.34435vw + 13.38843px)}.kep-login-facebook.metro{border-radius:0}.kep-login-facebook .fa{margin-right:calc(.34435vw + 3.38843px)}",
                "",
              ],
            ],
          }
        }
      />
    </span>
  </div>
</div>
`;
